### DISCUSSION TITLE
How to show image in table view?

### DISCUSSION THREAD

**Discussion by bartclaeys** [filamentphp/filament#5141] [Help] [answered]
I would like to show a thumbnail in my table list view, but the field in my database does not have the full path plus I need to add parameters towards the end of the URL (to dynamically resize the image, reduce quality), so I either need:

1. A closure I can modify, as part of `ImageColumn::make()` but I don't see any ImageColumn parameters to hook into.
2. Mutator that runs before showing the Table, but there does not seem to be a hook for `Table`.

What else can I attempt?

**Comment by ArtMin96** [comment]
Can you try this? I'm not sure if this is the right option, but you can try

```php
ImageColumn::make('img')
    ->getStateUsing(function (Model $record): string {
        return $record->image;
    })
```

**Reply by bartclaeys** [reply]
Thanks, but this results in two errors:

> Too few arguments to function Column::make(), 0 passed.

This error goes away by defining make(''). But then this error appears:

> Column extends [Filament\Support\Components\ViewComponent] but does not have a [$view] property defined.

**Reply by ArtMin96** [reply]
I mean `ImageColumn` not `Column`

**Reply by bartclaeys** [reply]
This works better, and prints out the following, but can't get an image in there :(

`<div style=" height: 40px;" class="">`

I tried the following but without success:

```
ImageColumn::make('img')
     ->getStateUsing(function (Book $record): string {
         return "<img src='" . $record->coverphoto . "'/>";
    }),
```

**Comment by bartclaeys** [comment]
Got it to work!

```
Tables\Columns\TextColumn::make('img')
        ->getStateUsing(function (Book $record): string {
        return "<img src='" . $record->coverphoto . "/'/>";
    })->html(),
```

**Reply by ArtMin96** [reply]
IMO this is not a good option. Why don't you just do this?
```php
ImageColumn::make('coverphoto')
    ->getStateUsing(function (Book $record): string {
        return $record->coverphoto.'your dynamic url';
    })
    ->extraImgAttributes([
        'attribute' => 'attr-value',
    ]),
```

**Reply by ArtMin96** [reply]
> IMO this is not a good option. Why don't you just do this?
> 
> ```
> ImageColumn::make('coverphoto')
>     ->getStateUsing(function (Book $record): string {
>         return $record->coverphoto.'your dynamic url';
>     })
>     ->extraImgAttributes([
>         'attribute' => 'attr-value',
>     ]),
> ```

Can you mark this as answered so others don't ask the same question again? Maybe one day I will need it too. :smiley:

**Comment by bartclaeys** [comment, marked as answer]
Yay, that works! I agree it's better than using a text column. 

```
ImageColumn::make('coverphoto')
    ->getStateUsing(function (Book $record): string {
        return $record->imgurl;
    })
    ->extraImgAttributes([
        'img' => 'src'
    ]),
```

**Reply by ArtMin96** [reply]
Cool. Now you can also use `ImageColumn` features ex. `->circular()` if you need

**Comment by VitoBryliano** [comment]
I saw a Filament\Tables\Columns\ImageColumn to describe image column
